src/AstriaOracle.sol

```diff
@@ -36,21 +46,37 @@ contract AstriaOracle {
     // occurs when attempting to update the price for an uninitialized currency pair
     error UninitializedCurrencyPair(uint256 index);
 
+    // occurs when attempting to initialize an unauthorized currency pair, when authorization is required
+    error UnauthorizedCurrencyPair(bytes32 currencyPair);
+
     modifier onlyOracle() {
         require(msg.sender == ORACLE, "AstriaOracle: only oracle can update");
         _;
     }
 
-    constructor(address _oracle) {
+    constructor(address _oracle, bool _requireCurrencyPairAuthorization) Ownable(msg.sender) {
         ORACLE = _oracle;
+        requireCurrencyPairAuthorization = _requireCurrencyPairAuthorization;
+    }
+
+    function setRequireCurrencyPairAuthorization(bool _requireCurrencyPairAuthorization) external onlyOwner {
+        requireCurrencyPairAuthorization = _requireCurrencyPairAuthorization;
+    }
+
+    function authorizeCurrencyPair(bytes32 _currencyPair) external onlyOwner {
+        authorizedCurrencyPairs[_currencyPair] = true;
     }
 
     function initializeCurrencyPair(bytes32 _currencyPair, uint8 _decimals) external onlyOracle {
+        if (requireCurrencyPairAuthorization && !authorizedCurrencyPairs[_currencyPair]) {
+            revert UnauthorizedCurrencyPair(_currencyPair);
+        }
```

Comment:
Having been initialized once implies authorization, the only way this happens authorization not set and initialized is previously authorization wasn't required. This allows not using up additional memory space when never needed. An alternative would be to always set authorization at time of setting initialized, but that isn't space efficient.

This ensures that you can update currencyPairs via reinitializing if they have already been set, enabling decimals to be updated.

```suggestion
        if (requireCurrencyPairAuthorization && !authorizedCurrencyPairs[_currencyPair] && !initializedCurrencyPairs[_currencyPair].initialized) {
            revert UnauthorizedCurrencyPair(_currencyPair);
        }
```